### DISCUSSION TITLE
Support rearrange blocks by drag and drop

### DISCUSSION THREAD

**Discussion by danny-laa** [outline/outline#1414] [General] [answered]
It would be nice if we can rearrange blocks by drag and drop.

![Draggy](https://user-images.githubusercontent.com/46824499/89194040-6c4fa580-d5d9-11ea-8edf-2202ba4b4326.gif)

![notion-columns](https://user-images.githubusercontent.com/46824499/89194222-ac168d00-d5d9-11ea-9a6e-cd539b6d4d3d.gif)

**Comment by tommoor** [comment, marked as answer]
Agreed, there are already multiple issues covering this request:

https://github.com/outline/outline/issues/1253
https://github.com/outline/rich-markdown-editor/issues/215

**Comment by matbrgz** [comment]
@tommoor is possible to implement blocks of texts/content? Like the second example?

**Reply by almereyda** [reply]
Horizontal stacking of blocks seems very interesting. The idea is close to what [TreeSheets](https://strlen.com/treesheets/) and [Gingko](https://gingkoapp.com/) do.

**Comment by Thomascountz** [comment]
I'd like to continue this idea 😄 

Today, to my knowledge, only list items (ordered, unordered, and checkboxes), images, files, page breaks, section breaks, are "drag-and-droppable." It would be nice if headings, code blocks, warnings, paragraphs, et al. were also moveable throughout the page via drag-and-drop.

Example of moving a paragraph in Notion (from above):
![](https://user-images.githubusercontent.com/46824499/89194222-ac168d00-d5d9-11ea-9a6e-cd539b6d4d3d.gif)

**Comment by matteoveglia** [comment]
Is there any movement on this? I see the issue linked in this comment has been closed 

> Agreed, there are already multiple issues covering this request:
> 
> #1253 [outline/rich-markdown-editor#215](https://github.com/outline/rich-markdown-editor/issues/215)

**Reply by tommoor** [reply]
No movement, improvements to drag and drop are always welcomed though

**Comment by EduNetEurope** [comment]
Requested again this year: https://github.com/outline/outline/discussions/6921

**Comment by laGameTV** [comment]
bump

**Comment by mary-loulou** [comment]
It is actually possible:
https://github.com/outline/outline/discussions/6921#discussioncomment-10044991

**Comment by seamon67** [comment]
Bumpity Bumpity Bump Bump Bump

**Comment by Matt-uSk** [comment]
Bumpity Bumpity Bump Bump Bump too ahaha
